projects/ngneat/transloco-locale/src/lib/helpers.ts

```diff
@@ -0,0 +1,70 @@
+import NumberFormatOptions = Intl.NumberFormatOptions;
+import { toNumber } from '@ngneat/transloco';
+import { Locale, DateFormatOptions } from './transloco-locale.types';
+
+export const ISO8601_DATE_REGEX = /^(\d{4})-?(\d\d)-?(\d\d)(?:T(\d\d)(?::?(\d\d)(?::?(\d\d)(?:\.(\d+))?)?)?(Z|([+-])(\d\d):?(\d\d))?)?$/;
+/**
+ * check if a given value is in BCP 47 language tag.
+ *
+ * isLocaleFormat('en') // false,
+ * isLocaleFormat('En-us') // false
+ * isLocaleFormat('en-US') // true
+ */
+export function isLocaleFormat(val: string): val is Locale {
+  return typeof val === 'string' && !!val.match(/[a-z]{2}-[A-Z]{2}/);
+}
+
+export function localizeNumber(value: number | string, locale: Locale, options: NumberFormatOptions): string {
+  const number = toNumber(value);
+  return number !== null ? new Intl.NumberFormat(locale, options).format(number) : '';
+}
+
+export function localizeDate(date: Date, locale: Locale, options: DateFormatOptions): string {
+  if (isDate(date)) {
+    return date.toLocaleDateString(locale, options);
+  }
+  return '';
+}
+
+export function isDate(value): boolean {
+  return value instanceof Date && !isNaN(<any>value);
+}
+
+export function toDate(value): Date {
+  let match: RegExpMatchArray | null;
+
+  if (typeof value === 'string') {
+    value = value.trim();
+  }
+  if (isDate(value)) {
+    return value;
+  } else if (!isNaN(value - parseFloat(value))) {
+    return new Date(parseFloat(value));
+  } else if (typeof value === 'string' && /^(\d{4}-\d{1,2}-\d{1,2})$/.test(value)) {
+    const [y, m, d] = value.split('-').map((val: string) => parseInt(val, 10));
+    return new Date(y, m - 1, d);
+  } else if (typeof value === 'string' && (match = value.match(ISO8601_DATE_REGEX))) {
+    return isoStringToDate(match);
+  } else {
+    return new Date(value);
+  }

```

Comment:
Not very important, but each conditional check already returns early in this function. Not using `else if` or `else` would make it even more readable. (Most brackets are omittable here.)

```
export function toDate(value): Date {
  let match: RegExpMatchArray | null;

  if (typeof value === 'string') {
    value = value.trim();
  }

  if (isDate(value)) {
    return value;
  }
  
  if (!isNaN(value - parseFloat(value))) {
    return new Date(parseFloat(value));
  }
  
  if (typeof value === 'string' && /^(\d{4}-\d{1,2}-\d{1,2})$/.test(value)) {
    const [y, m, d] = value.split('-').map((val: string) => parseInt(val, 10));
    return new Date(y, m - 1, d);
  }

  if (typeof value === 'string' && (match = value.match(ISO8601_DATE_REGEX))) {
    return isoStringToDate(match);
  }
  
  return new Date(value);
}
```